aten/src/ATen/cpu/vec256/vsx/vec256_complex_float_vsx.h

```diff
@@ -0,0 +1,686 @@
+    auto z = Vec256<ComplexFlt>(zero);
+    return z - *this;
+  }
+  Vec256<ComplexFlt> round() const {
+    return {vec_round(_vec0), vec_round(_vec1)};
+  }
+  Vec256<ComplexFlt> tan() const {
+    return map(std::tan);
+  }
+  Vec256<ComplexFlt> tanh() const {
+    return map(std::tanh);
+  }
+  Vec256<ComplexFlt> trunc() const {
+    return {vec_trunc(_vec0), vec_trunc(_vec1)};
+  }
+
+  Vec256<ComplexFlt> elwise_sqrt() const {
+    return {vec_sqrt(_vec0), vec_sqrt(_vec1)};
+  }
+
+  void dump() const {
+    std::cout << _vec0[0] << "," << _vec0[1] << "," << _vec0[2] << ","
+              << _vec0[3] << ",";
+    std::cout << _vec1[0] << "," << _vec1[1] << "," << _vec1[2] << ","
+              << _vec1[3] << std::endl;
+  }
+
+  Vec256<ComplexFlt> sqrt() const {
+    //   sqrt(a + bi)
+    // = sqrt(2)/2 * [sqrt(sqrt(a**2 + b**2) + a) + sgn(b)*sqrt(sqrt(a**2 +
+    // b**2) - a)i] = sqrt(2)/2 * [sqrt(abs() + a) + sgn(b)*sqrt(abs() - a)i]
+
+    auto sign = *this & isign_mask;
+    auto factor = sign | sqrt2_2;
+    auto a_a = el_mergee();
+    // a_a.dump();
+    a_a = a_a ^ isign_mask; // a -a
+    auto res_re_im =
+        (abs_() + a_a).elwise_sqrt(); // sqrt(abs + a) sqrt(abs - a)
+    return factor.elwise_mult(res_re_im);
+  }
+
+  Vec256<ComplexFlt> reciprocal() const {
+    // re + im*i = (a + bi)  / (c + di)
+    // re = (ac + bd)/abs_2() = c/abs_2()
+    // im = (bc - ad)/abs_2() = d/abs_2()
+    auto c_d = *this ^ isign_mask; // c       -d
+    auto abs = abs_2_();
+    return c_d.elwise_div(abs);
+  }
+
+  Vec256<ComplexFlt> rsqrt() const {
+    return sqrt().reciprocal();
+  }
+
+  Vec256<ComplexFlt> pow(const Vec256<ComplexFlt>& exp) const {
+    __at_align32__ ComplexFlt x_tmp[size()];
+    __at_align32__ ComplexFlt y_tmp[size()];
+    store(x_tmp);
+    exp.store(y_tmp);
+    for (int i = 0; i < size(); i++) {
+      x_tmp[i] = std::pow(x_tmp[i], y_tmp[i]);
+    }
+    return loadu(x_tmp);
+  }
+
+  Vec256<ComplexFlt> atan() const {
+    // atan(x) = i/2 * ln((i + z)/(i - z))
+    auto ione = Vec256(imag_one);
+    auto sum = ione + *this;
+    auto sub = ione - *this;
+    auto ln = (sum / sub).log(); // ln((i + z)/(i - z))
+    return ln * imag_half; // i/2*ln()
+  }
+
+  Vec256<ComplexFlt> acos() const {
+    // acos(x) = pi/2 - asin(x)
+    return Vec256(pi_2) - asin();
+  }
+
+  Vec256<ComplexFlt> inline operator*(const Vec256<ComplexFlt>& b) const {
+    //(a + bi)  * (c + di) = (ac - bd) + (ad + bc)i
+
+#if 1
+    // this is more vsx friendly than simulating horizontall from x86
+
+    auto vi = b.el_mergeo();
+    auto vr = b.el_mergee();
+    vi = vi ^ rsign_mask;
+    auto ret = elwise_mult(vr);
+    auto vx_swapped = el_swapped();
+    ret = vx_swapped.el_madd(vi, ret);
+    return ret;
+
+#else
+
+    auto ac_bd = elwise_mult(b);
+    auto d_c = b.el_swapped();
+    d_c = d_c ^ isign_mask;
+    auto ad_bc = elwise_mult(d_c);
+    auto ret = horizontal_sub_permD8(ac_bd, ad_bc);
+    return ret;
```

Comment:
Are these another comment?

Reply:
The second one immitates x86 avx2. But I later switched to faster vsx one. 
As with floats fma and order affects precision. But I decided to test against arch related implementations.

Reply:
I kept it for now

Reply:
OK. Would you remove this before merger, or you would keep it even after merger? If so, could you add more comments as why here's `#if 1`?